pkg/apis/workflow.go

```diff
@@ -102,3 +102,266 @@ func CreateWorkflow(requestData model.ChaosWorkFlowRequest, cred types.Credentia
+			Token:    cred.Token,
+		},
+		query,
+		string(types.Post),
+	)
+	if err != nil {
+		return WorkflowRunsListData{}, err
+	}
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return WorkflowRunsListData{}, err
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		var workflowRunsList WorkflowRunsListData
+		err = json.Unmarshal(bodyBytes, &workflowRunsList)
+		if err != nil {
+			return WorkflowRunsListData{}, err
+		}
+
+		if len(workflowRunsList.Errors) > 0 {
+			return WorkflowRunsListData{}, errors.New(workflowRunsList.Errors[0].Message)
+		}
+
+		return workflowRunsList, nil
+	} else {
+		return WorkflowRunsListData{}, err
+	}
+}
+
+type DeleteChaosWorkflowData struct {
+	Errors []struct {
+		Message string   `json:"message"`
+		Path    []string `json:"path"`
+	} `json:"errors"`
+	Data DeleteChaosWorkflowDetails `json:"data"`
+}
+
+type DeleteChaosWorkflowDetails struct {
+	IsDeleted bool `json:"deleteChaosWorkflow"`
+}
+
+type DeleteChaosWorkflowGraphQLRequest struct {
+	Query     string `json:"query"`
+	Variables struct {
+		ProjectID     string  `json:"projectID"`
+		WorkflowID    *string `json:"workflowID"`
+		WorkflowRunID *string `json:"workflowRunID"`
+	} `json:"variables"`
+}
+
+// DeleteChaosWorkflow sends GraphQL API request for deleting a given Chaos Workflow.
+func DeleteChaosWorkflow(projectID string, workflowID *string, cred types.Credentials) (DeleteChaosWorkflowData, error) {
+
+	var gqlReq DeleteChaosWorkflowGraphQLRequest
+	var err error
+
+	gqlReq.Query = `mutation deleteChaosWorkflow($projectID: String!, $workflowID: String, $workflowRunID: String) {
+                      deleteChaosWorkflow(
+                        projectID: $projectID
+                        workflowID: $workflowID
+                        workflowRunID: $workflowRunID
+                      )
+                    }`
+	gqlReq.Variables.ProjectID = projectID
+	gqlReq.Variables.WorkflowID = workflowID
+	var workflow_run_id string = ""
+	gqlReq.Variables.WorkflowRunID = &workflow_run_id
+
+	query, err := json.Marshal(gqlReq)
+	if err != nil {
+		return DeleteChaosWorkflowData{}, err
+	}
+
+	resp, err := SendRequest(
+		SendRequestParams{
+			Endpoint: cred.Endpoint + utils.GQLAPIPath,
+			Token:    cred.Token,
+		},
+		query,
+		string(types.Post),
+	)
+	if err != nil {
+		return DeleteChaosWorkflowData{}, err
+	}
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return DeleteChaosWorkflowData{}, err
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		var deletedWorkflow DeleteChaosWorkflowData
+		err = json.Unmarshal(bodyBytes, &deletedWorkflow)
+		if err != nil {
+			return DeleteChaosWorkflowData{}, err
+		}
+
+		if len(deletedWorkflow.Errors) > 0 {
+			return DeleteChaosWorkflowData{}, errors.New(deletedWorkflow.Errors[0].Message)
+		}
+
+		return deletedWorkflow, nil
+	} else {
+		return DeleteChaosWorkflowData{}, err
+	}
```

Comment:
Here also we can return custom error message instead of directly returning error.